python/tests/res/fm/test_ecl_run_new.py

```diff
@@ -0,0 +1,197 @@
+import inspect
+import yaml
+import pytest
+import re
+
+from ecl.summary import EclSum
+from ecl.util.test import TestAreaContext
+from tests import ResTest
+from tests.utils import tmpdir
+from res.fm.ecl import *
+from subprocess import Popen, PIPE
+from subprocess import Popen, PIPE
+from distutils.spawn import find_executable
+from _pytest.monkeypatch import MonkeyPatch
+
+
+def find_version(output):
+    return re.search(r"flow\s*([\d.]+)", output).group(1)
+
+
+class EclRunTest(ResTest):
+    def setUp(self):
+        self.ecl_config_path = os.path.dirname(inspect.getsourcefile(Ecl100Config))
+        self.monkeypatch = MonkeyPatch()
+
+    def tearDown(self):
+        self.monkeypatch.undo()
+
+    @staticmethod
+    def _eclrun_conf():
+        return {
+            "eclrun_env": {
+                "SLBSLS_LICENSE_FILE": "7321@eclipse-lic-no.statoil.no",
+                "ECLPATH": "/prog/res/ecl/grid",
+                "PATH": "/prog/res/ecl/grid/macros",
+                "F_UFMTENDIAN": "big",
+            }
+        }
+
+    def init_eclrun_config(self):
+        conf = EclRunTest._eclrun_conf()
+        with open("ecl100_config.yml", "w") as f:
+            f.write(yaml.dump(conf))
+        self.monkeypatch.setenv("ECL100_SITE_CONFIG", "ecl100_config.yml")
+
+    @tmpdir()
+    @pytest.mark.equinor_test
+    def test_run(self):
+        self.init_eclrun_config()
+        shutil.copy(
+            os.path.join(self.SOURCE_ROOT, "test-data/local/eclipse/SPE1.DATA"),
+            "SPE1.DATA",
+        )
+        ecl_config = Ecl100Config()
+
+        ecl_run = EclRun("SPE1.DATA", None)
+        ecl_run.runEclipse(eclrun_config=EclrunConfig(ecl_config, "2019.3"))
+
+        ok_path = os.path.join(ecl_run.runPath(), "{}.OK".format(ecl_run.baseName()))
+        log_path = os.path.join(ecl_run.runPath(), "{}.LOG".format(ecl_run.baseName()))
+
+        self.assertTrue(os.path.isfile(ok_path))
+        self.assertTrue(os.path.isfile(log_path))
+        self.assertTrue(os.path.getsize(log_path) > 0)
+
+        errors = ecl_run.parseErrors()
+        self.assertEqual(0, len(errors))
+
+    @pytest.mark.equinor_test
+    @tmpdir()
+    def test_run_api(self):
+        self.init_eclrun_config()
+        shutil.copy(
+            os.path.join(self.SOURCE_ROOT, "test-data/local/eclipse/SPE1.DATA"),
+            "SPE1.DATA",
+        )
+        ecl_config = Ecl100Config()
+        run(ecl_config, ["SPE1.DATA", "--version=2019.3"])
+
+        self.assertTrue(os.path.isfile("SPE1.DATA"))
+
+    @pytest.mark.equinor_test
+    @tmpdir()
+    def test_failed_run(self):
+        self.init_eclrun_config()
+        shutil.copy(
+            os.path.join(self.SOURCE_ROOT, "test-data/local/eclipse/SPE1_ERROR.DATA"),
+            "SPE1_ERROR.DATA",
+        )
+        ecl_config = Ecl100Config()
+        eclrun_config = EclrunConfig(ecl_config, "2019.3")
+        ecl_run = EclRun("SPE1_ERROR", None)
+        with self.assertRaises(Exception):
```

Comment:
No need to do a double run, keep the context instead:

```
with self.assertRaises(Exception) as error_context:
    ecl_run.runEclipse(eclrun_config=eclrun_config)
self.assertIn("ERROR", str(error_context.exception))
```

There are multiple checks further down in the tests here that might benefit from checking the error string, as they do assert on fairly generic errors.